alpha/declcfg/filter/config/v1alpha1/config.go

```diff
@@ -0,0 +1,77 @@
+package v1alpha1
+
+import (
+	"errors"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+// FilterConfigurationV1 is a configuration for filtering a set of packages and channels from a catalog.
+// It supports selecting specific packages and specific channels and/or versions within those packages.
+// The configuration is intended to be used with the `opm render` command to generate a filtered catalog.
+type FilterConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Packages is a list of packages to include in the filtered catalog.
+	Packages []Package `json:"packages"`
+}
+
+type Package struct {
+	// Name is the name of the package to filter.
+	Name string `json:"name"`
+
+	// DefaultChannel is the new default channel to use for the package.
+	// If not set, the default channel will be the same as the original default channel.
+	// If the original default channel is not in the filtered catalog, this field must be set.
+	DefaultChannel string `json:"defaultChannel,omitempty"`
+
+	// Channels is a list of channels to include in the filtered catalog.
+	// If not set, all channels will be included.
+	Channels []Channel `json:"channels,omitempty"`
+}
+
+type Channel struct {
+	// Name is the name of the channel to include in the filtered catalog.
+	Name string `json:"name"`
+
+	// VersionRange is a semver range to filter the versions of the channel.
+	// If not set, all versions will be included.
+	VersionRange string `json:"versionRange,omitempty"`
+}
+
+func LoadFilterConfiguration(data []byte) (*FilterConfiguration, error) {
+	cfg := &FilterConfiguration{}
+	if err := yaml.Unmarshal(data, cfg); err != nil {
```

Comment:
Can this `yaml.Unmarshal` handle both YAML and JSON formatted data? If not, a go doc that the input data _has_ to be valid YAML would be nice.

Reply:
Yes, this supports both.